stdlib/stack_trace.mli

```diff
@@ -0,0 +1,141 @@
+    trace. *)
+
+val length: t -> int
+(** [length st] returns the number of (non-inlined) slots in the stack trace
+    [st]. *)
+
+module Slot : sig
+  (** {1 Manipulation of stack trace information}
+
+      These functions are used to traverse the slots of a stack trace and
+      extract information from them in a programmer-friendly format. *)
+
+  type stack_trace
+
+  type t
+  (** The abstract type [t] represents a single slot of a stack trace. *)
+
+  val is_raise: t -> bool
+  (** [is_raise slot] is [true] when [slot] refers to a raising point in the
+      code, and [false] when it comes from a simple function call. *)
+
+  val is_inline: t -> bool
+  (** [is_inline slot] is [true] when [slot] refers to a call that got inlined
+      by the compiler, and [false] when it comes from any other context. *)
+
+  type location =
+    { filename: string;
+      line_number: int;
+      start_char: int;
+      end_char: int }
+  (** The type of location information found in stack traces. [start_char] and
+      [end_char] are positions relative to the beginning of the line. *)
+
+  val location: t -> location option
+  (** [location slot] returns the location information of the slot,
+      if available, and [None] otherwise.
+
+      Some possible reasons for failing to return a location are as follow:
+      - the slot corresponds to a compiler-inserted raise
+      - the slot corresponds to a part of the program that has not been
+      compiled with debug information ([-g]) *)
+
+  val format: int -> t -> string option
+  (** [format pos slot] returns the string representation of [slot] as
+      {!to_string} would format it, assuming it is the [pos]-th element of the
+      callstack: the [0]-th element is pretty-printed differently than the
+      others.
+
+      Whole-stack trace printing functions also skip some uninformative slots;
+      in that case, [format pos slot] returns [None]. *)
+
+  module Raw : sig
+    (** {1 Raw stack trace slots} *)
+
+    type t
+    (** This type allows direct access to raw stack trace slots, without any
+        conversion in an OCaml-usable data-structure. Being process-specific,
+        they must absolutely not be marshalled, and are unsafe to use for this
+        reason (marshalling them may not fail, but un-marshalling and using the
+        result will result in undefined behavior).
+
+        Elements of this type can still be compared and hashed: when two
+        elements are equal, then they represent the same source location (the
+        converse is not necessarily true in presence of inlining, for
+        example). *)
+
+    val get: stack_trace -> int -> t
+    (** [get st pos] returns the slot in position [pos] in the stack trace
+        [st]. *)
+
+    val next: t -> t option
+    (** [next slot] returns the next inlined slot, if any.
+
+        Sample code to iterate over all frames (inlined and non-inlined):
```

Comment:
Note that "frames" is used here instead of "slots".

Reply:
Thanks, fixed the docs to use "frame(s)" everywhere.